src/systems/a320_systems/src/pneumatic.rs

```diff
@@ -1,27 +1,3039 @@
+
+        let (left_system, right_system) = self.engine_systems.split_at_mut(1);
+        self.apu_bleed_air_valve
+            .update_move_fluid(context, &mut self.apu, &mut left_system[0]);
+
+        self.cross_bleed_valve.update_move_fluid(
+            context,
+            &mut left_system[0],
+            &mut right_system[0],
+        );
+
+        self.green_hydraulic_reservoir_with_valve
+            .update_flow_through_valve(context, &mut left_system[0]);
+        self.blue_hydraulic_reservoir_with_valve
+            .update_flow_through_valve(context, &mut left_system[0]);
+        self.yellow_hydraulic_reservoir_with_valve
+            .update_flow_through_valve(context, &mut left_system[0]);
+
+        self.packs[0].update(context, &mut left_system[0]);
+        self.packs[1].update(context, &mut right_system[0]);
+    }
+
+    // TODO: Returning a mutable reference here is not great. I was running into an issue with the update order:
+    // - The APU turbine must know about the bleed valve being open as soon as possible to update EGT properly
+    // - To open the bleed valve, we need a signal from the ecb
+    // - To get a signal from the ECB to open the bleed valve, we have to update the APU.
+    // For now, we just pass over control of the bleed valve to the APU, so it can be updated after the ECB update but before the turbine update.
+    pub fn apu_bleed_air_valve(&mut self) -> &mut impl ControllablePneumaticValve {
+        &mut self.apu_bleed_air_valve
+    }
+
+    fn update_hydraulic_reservoir_spatial_volumes(
+        &mut self,
+        green_hydraulic_reservoir_volume: Volume,
+        blue_hydraulic_reservoir_volume: Volume,
+        yellow_hydraulic_reservoir_volume: Volume,
+    ) {
+        self.green_hydraulic_reservoir_with_valve
+            .container()
+            .change_spatial_volume(green_hydraulic_reservoir_volume);
+        self.blue_hydraulic_reservoir_with_valve
+            .container()
+            .change_spatial_volume(blue_hydraulic_reservoir_volume);
+        self.yellow_hydraulic_reservoir_with_valve
+            .container()
+            .change_spatial_volume(yellow_hydraulic_reservoir_volume);
+    }
+}
+impl SimulationElement for A320Pneumatic {
+    fn accept<T: SimulationElementVisitor>(&mut self, visitor: &mut T) {
+        self.apu_bleed_air_controller.accept(visitor);
+        self.cross_bleed_valve.accept(visitor);
+        self.fadec.accept(visitor);
+
+        accept_iterable!(self.bleed_monitoring_computers, visitor);
+        accept_iterable!(self.engine_systems, visitor);
+        accept_iterable!(self.packs, visitor);
+
+        visitor.visit(self);
+    }
+
+    fn write(&self, writer: &mut SimulatorWriter) {
+        writer.write(
+            &self.cross_bleed_valve_open_id,
+            self.cross_bleed_valve.is_open(),
+        );
+        writer.write(
+            &self.apu_bleed_air_valve_open_id,
+            self.apu_bleed_air_valve.is_open(),
+        );
```

Comment:
**thought**: Would it maybe make sense to move these writes into the types themselves?

Reply:
I use a `DefaultValve` for the APU bleed air valve and I don't really want to inject a variable name into that struct, just so it can write its state. For consistency, I will also refrain from doing that for the cross bleed valve.